cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/AbstractMapReduceContextBuilder.java

```diff
@@ -106,7 +107,7 @@ public BasicMapReduceContext build(MapReduceMetrics.TaskType type,
     BasicMapReduceContext context =
       new BasicMapReduceContext(program, type, RunIds.fromString(runId), taskId,
                                 runtimeArguments, datasets, spec, logicalStartTime,
-                                workflowBatch, discoveryServiceClient, metricsCollectionService,
+                                workflowBatch, adapterName, discoveryServiceClient, metricsCollectionService,
```

Comment:
Adapter is not concept known to MapReduce right ? If we make it that, what happens when it's not associated with Adapter, but used as is in more general terms.